harness/determined/core/_context.py

```diff
@@ -189,7 +199,15 @@ def init(
 
     storage_manager = None
     if checkpoint_storage is not None:
-        storage_manager = storage.from_string(checkpoint_storage)
+        if isinstance(checkpoint_storage, str):
```

Comment:
Is there a way to write the code so that there is only one instance of this logic?

Having user-input validation in two different places is an especially-dangerous type of code duplication, because it's so obvious to end-users if there's ever a divergence.